universal-application-tool-0.0.1/app/services/Path.java

```diff
@@ -175,8 +184,7 @@ private String keyNameWithoutArrayIndex() {
           .replace(matcher.start(ARRAY_SUFFIX_GROUP), matcher.end(ARRAY_SUFFIX_GROUP), "")
           .toString();
     }
-    throw new IllegalStateException(
-        String.format("This path %s does not reference an array element.", this));
+    return keyName();
```

Comment:
(blocking): can this be removed if the checkbox question path has "[]" at the end?

Reply:
I kept Path as-is and chose to handle this in `BlockDefinition` instead